### ISSUE TITLE
vgg16 predictions: class probabilities change significantly in 2.0.9

### ISSUE THREAD

**Issue by jjallaire** [fchollet/keras#8426]
Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

I've noticed when updating to Keras 2.0.9 that the class probabilities for predictions on the "creative_commons_elephant.jpg" image used in tutorials change significantly (i.e. African_elephant goes from `0.909` to `0.789`).

Here is my script:

```python
from keras.applications.vgg16 import VGG16
model = VGG16(weights='imagenet')

from keras.preprocessing import image
from keras.applications.vgg16 import preprocess_input, decode_predictions
import numpy as np

img_path = 'creative_commons_elephant.jpg'
img = image.load_img(img_path, target_size=(224, 224))
x = image.img_to_array(img)
x = np.expand_dims(x, axis=0)
x = preprocess_input(x)

preds = model.predict(x)
print('Predicted:', decode_predictions(preds, top=3)[0])
```

Under Keras 2.0.8 this gives:

```
('Predicted:', [(u'n02504458', u'African_elephant', 0.90942073), (u'n01871265', u'tusker', 0.086183183), (u'n02504013', u'Indian_elephant', 0.0043545808)])
```

However under Keras 2.0.9 it gives:

```
('Predicted:', [(u'n02504458', u'African_elephant', 0.78988522), (u'n01871265', u'tusker', 0.1987267), (u'n02504013', u'Indian_elephant', 0.011142471)])
```

**Comment by fchollet** [comment]
I believe `load_img` has changed. Try to test your script with the version of `load_img` that was in 2.0.8?

**Comment by fchollet** [comment]
I would also point out that african elephant, tusker and indian elephant all look pretty similar, so it's not necessarily strange that very small changes in inputs could result in relatively significant probability shuffling between these 3 classes.

**Comment by jjallaire** [comment]
Yes indeed substituting the definition of `load_img` from 2.0.8 results in 2.0.9 giving the same predictions as it previously did.

Realize that this might not necessarily be a bug but just wanted you to be aware of the difference in the case that it is.

**Comment by fchollet** [comment]
I think we should make sure that the default behavior of `load_img` is the same as in 2.0.8. We'll do a 2.1.0 release very soon anyway. Please open a PR if you can identify where exactly the issue lies.

**Comment by jjallaire** [comment]
It looks to me as if img.resize uses "nearest" as it's default if not
specified (seehttps://
pillow.readthedocs.io/en/4.3.x/reference/Image.html#PIL.Image.Image.resize)
whereas we use "bilinear" as the default in Keras 2.0.9. So it seems like
the source of the difference is the change in default resizing from
"nearest" to "bilinear".
